config_converter/config_mapper/config_mapper.py

```diff
@@ -0,0 +1,189 @@
+    # these dicts can be updated when more plugins are supported
+    for d in config_obj.directives:
+        if d.name not in plugin_prefix_map:
+            print("Currently we do not support plugins of " + d.name)
+            continue
+        for p in d.params:
+            if p.name == '@type':
+                # identify the plugin
+                plugin = plugin_prefix_map[d.name] + p.value
+                if plugin in _SUPPORTED_PLUGINS:
+                    result['logs_module'][dir_name_map[d.name]] = \
+                            result['logs_module'].get(
+                                dir_name_map[d.name], [])
+                    converted_dir: dict = _plugin_convert(d, plugin)
+                    result['logs_module'][dir_name_map[d.name]]\
+                                .append(converted_dir)
+                elif plugin in _UNSUPPORTED_PLUGINS:
+                    print("we dont support plugin " + plugin)
+                else:
+                    print("we dont know plugin " + plugin)
+    return result
+
+
+def _plugin_convert(d: config_pb2.Directive, plugin: str) -> dict:
+    """Uses mapping rules passed to function to map all fields."""
+    cur = {}
+    outer_map = _OUTER_MAP[d.name]
+    main_map = _MAIN_MAP[plugin]
+    special_params = _SPECIAL_MAP[plugin]
+    supported_dirs = _SUPPORTED_DIRS[plugin]
+    plugin_type_map = {'tail': 'file'}
+    outer_map_copy = copy.copy(outer_map)
+    # ensure all outer level params needed are there in directive
+    for p in d.params:
+        if p.name in outer_map_copy:
+            if p.name == '@type':
+                cur[outer_map[p.name]] = plugin_type_map[p.value]
+            else:
+                cur[outer_map[p.name]] = p.value
+            outer_map_copy.pop(p.name)
+    if outer_map_copy != {}:
+        print('invalid configuration')
+        sys.exit()
+    specific = {}
+    for p in d.params:
+        if p.name not in outer_map:
+            if p.name in _UNSUPPORTED_FIELDS:
+                print(f'parameter {p.name} cant be converted')
+            elif p.name in special_params:
+                _map_special_fields(specific, p, plugin)
+            elif p.name not in main_map:
+                print(f'parameter {p.name} is unknown')
+            else:
+                specific[main_map[p.name]] = p.value
+    for nd in d.directives:
+        if nd.name not in supported_dirs:
+            print(f'directive {nd.name} is not supported')
+        else:
+            for np in nd.params:
+                _map_special_fields(specific, np, plugin)
+    cur[f'{cur["type"]}_{d.name}_config'] = specific
+    return cur
+
+
+def _map_special_fields(specific: dict, p: config_pb2.Param,
+                        plugin: str) -> None:
+    """Cases on p with all special_map dict, calls a map function."""
+    if p.name in _SPECIAL_MAP[plugin]:
+        if plugin == 'in_tail':
+            _parser_dir(specific, p)
```

Comment:
I believe this function does the mapping (or assign the mappings) for the special fields parser - is it reasonable to call it `_map_parser_dir` - basically I think if this function starts with a verb, that will make it more clear.

Reply:
Good suggestion! I will change that.